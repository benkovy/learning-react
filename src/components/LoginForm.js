import React from 'react';

export default class LoginForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            type: 'login',
            error: ''
        };
    }

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({ email }));
    };

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    };

    onTypeChange = (e) => {
        const type = e.target.value.toLowerCase();
        this.setState(() => ({ type }));
    };

    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.email || !this.state.password) {
            this.setState(() => ({ error: 'Please enter a valid email and password' }));
        } else {
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                type: this.state.type,
                email: this.state.email,
                password: this.state.password
            })
        }
    };

    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text"
                        placeholder="Email"
                        autoFocus
                        value={this.state.email}
                        onChange={this.onEmailChange}
                    />
                    <input
                        type="text"
                        placeholder="Password"
                        value={this.state.password}
                        onChange={this.onPasswordChange}
                    />
                    <button>Go</button>
                    <select
                        value={this.state.type}
                        onChange={this.onTypeChange}>
                        <option value="login">Login</option>
                        <option value="create">Create</option>
                    </select>
                </form>
            </div>
        );
    }
}