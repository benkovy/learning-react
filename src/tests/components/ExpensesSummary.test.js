import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should correctly render expenses summary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expenses summary with many expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={22} expensesTotal={23512311} />);
    expect(wrapper).toMatchSnapshot();
});