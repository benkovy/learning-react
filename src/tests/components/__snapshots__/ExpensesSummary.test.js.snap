// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses summary with many expenses 1`] = `
<div>
  <h1>
    Viewing 
    22
     
    expenses
     totalling 
    $235,123.11
  </h1>
</div>
`;

exports[`Should correctly render expenses summary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;
